pages/_app.tsx

```diff
@@ -25,6 +25,7 @@ const App: AppType<AppProps> = ({ Component, pageProps }) => {
       site_list: pageProps.sites ?? [],
     },
   };
+  const hideNav = (Component as any).hideNav || false;
```

Comment:
<img width="423" alt="image" src="https://github.com/openclimatefix/pv-sites-mobile/assets/80798381/f8f56d47-b480-41dd-9117-e8a372296c42">

This is actually something I tried before using as any, but I'm getting this Typescript error: 
Property 'hideNav' does not exist on type 'NextComponentType<NextPageContext, any, any>'.
Property 'hideNav' does not exist on type 'ComponentClass<any, any> & { getInitialProps?(context: NextPageContext): any; }'.ts(2339)

Reply:
At the end of the day the only type safety we're doing here is ensuring there's no typo in `hideNav` on line 28 specifically, since this won't be recognized by the page files that actually _set_ `hideNav`. For this reason, the `any` here doesn't seem too big of a deal, and it'll take some more annoying typing to get this to check without `any`.

Reply:
@AndrewLester That makes sense, let me know if I should merge!